src/testFixtures/java/org/opensearch/knn/TestUtils.java

```diff
@@ -181,6 +186,45 @@ public static double calculateRecallValue(List<List<String>> searchResults, List
         return sum / recalls.size();
     }
 
+    public static PriorityQueue<DistVector> computeGroundTruthValues(int k, SpaceType spaceType, IDVectorProducer idVectorProducer) {
+        PriorityQueue<DistVector> pq = new PriorityQueue<>(k, new DistComparator());
+        int numDocs = idVectorProducer.getVectorCount();
+        float[] queryVector = idVectorProducer.getVector(numDocs);
+
+        for (int id = 0; id < numDocs; id++) {
+            float[] indexVector = idVectorProducer.getVector(id);
+            float dist = computeDistFromSpaceType(spaceType, indexVector, queryVector);
+            pq = sortVectorsUsingDist(pq, k, dist, id);
+        }
+        return pq;
+    }
+
+    public static float computeDistFromSpaceType(SpaceType spaceType, float[] indexVector, float[] queryVector) {
+        float dist = 0.0f;
+        if (spaceType != null) {
+            dist = KNN_SCORING_SPACE_TYPE.getOrDefault(
+                spaceType,
+                (defaultQueryVector, defaultIndexVector) -> throwInvalidInputException(spaceType)
+            ).apply(queryVector, indexVector);
+        }
+        return dist;
+    }
+
+    public static Float throwInvalidInputException(SpaceType spaceType) {
+        throw new IllegalArgumentException(String.format("Invalid SpaceType function: \"%s\"", spaceType));
+
+    }
+
+    public static PriorityQueue<DistVector> sortVectorsUsingDist(PriorityQueue<DistVector> pq, int k, float dist, int id) {
```

Comment:
sort is a misnomer. It is really adding and removing if necessary. Lucene has a similar function: https://github.com/apache/lucene/blob/main/lucene/core/src/java/org/apache/lucene/util/hnsw/NeighborQueue.java#L88. I would name similar to this.